Fauna/Configuration.cs

```diff
@@ -68,6 +74,11 @@ public Configuration(string secret = "", HttpClient? httpClient = null, ILogger?
             HttpClient = httpClient;
             DisposeHttpClient = false;
         }
+        else
+        {
+            // Set Timeout to int.MaxValue milliseconds and configure each SendAsync with a timebound CancellationToken
+            HttpClient = new HttpClient { Timeout = TimeSpan.FromMilliseconds(int.MaxValue) };
```

Comment:
https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.timeout?view=net-8.0#remarks

Don't accept this directly, there might be a namespace or import required.

```suggestion
            HttpClient = new HttpClient { Timeout = InfiniteTimeSpan };
```